include/gsl/gsl-lite.hpp

```diff
@@ -2672,6 +2672,7 @@ not_null<T> operator+( std::ptrdiff_t, not_null<T> const & ) gsl_is_delete;
 
 // not_null comparisons
 
+#if gsl_HAVE(TYPE_TRAITS)
```

Comment:
You can't just disable the comparison operators if `<type_traits>` is not available; that breaks a lot of code.